packages/components/grid/src/view-model.ts

```diff
@@ -193,12 +193,12 @@ export class GridViewModel<T = any> {
   }
 
   #getHeaderRows(columns: Array<GridColumn<T>>): Array<Array<GridColumn<T>>> {
-    const children = columns
-      .filter((col): col is GridColumnGroup<T> => col instanceof GridColumnGroup)
-      .reduce((acc: Array<Array<GridColumn<T>>>, cur) => {
-        return [...acc, ...this.#getHeaderRows(cur.columns)];
-      }, []);
-
-    return children.length ? [[...columns], children.flat(2)] : [[...columns]];
+    const groups = columns.filter((col): col is GridColumnGroup<T> => col instanceof GridColumnGroup);
+    const groupsNew = columns.map(col => (col instanceof GridColumnGroup ? col : new GridColumnGroup<T>()));
+    const columnsOutsideGroups = columns.filter((col): col is GridColumn<T> => !(col instanceof GridColumnGroup));
+    const children = groups.reduce((acc: Array<Array<GridColumn<T>>>, cur) => {
+      return [...acc, ...this.#getHeaderRows(cur.columns)];
+    }, []);
+    return children.length ? [[...groupsNew], [...children.flat(2), ...columnsOutsideGroups]] : [[...columns]];
```

Comment:
Can you explain what is fixed here? Perhaps add a comment explaining what is happening here? (I have to admit i also have a hard time understanding my own code)

Reply:
When you want a combination of grouped and ungrouped columns it didn't work. (I put it in the commitmessage, that's why it's handy to something other than emoji there 😜)